lib/src/editorhelper.cpp

```diff
@@ -234,4 +235,255 @@ std::unique_ptr<uhdr_raw_image_ext_t> apply_resize(uhdr_raw_image_t* src, int ds
+
+    uint32_t* rgbData = static_cast<uint32_t*>(src->planes[UHDR_PLANE_PACKED]);
+    unsigned int srcStride = src->stride[UHDR_PLANE_PACKED];
+
+    uint16_t* yData = static_cast<uint16_t*>(dst->planes[UHDR_PLANE_Y]);
+    uint16_t* uData = static_cast<uint16_t*>(dst->planes[UHDR_PLANE_UV]);
+    uint16_t* vData = uData + 1;
+
+    for (size_t i = 0; i < dst->h; i += 2) {
+      for (size_t j = 0; j < dst->w; j += 2) {
+        float r0 = float(rgbData[srcStride * i + j] & 0x3ff);
+        float g0 = float((rgbData[srcStride * i + j] >> 10) & 0x3ff);
+        float b0 = float((rgbData[srcStride * i + j] >> 20) & 0x3ff);
+
+        float r1 = float(rgbData[srcStride * i + j + 1] & 0x3ff);
+        float g1 = float((rgbData[srcStride * i + j + 1] >> 10) & 0x3ff);
+        float b1 = float((rgbData[srcStride * i + j + 1] >> 20) & 0x3ff);
+
+        float r2 = float(rgbData[srcStride * (i + 1) + j] & 0x3ff);
+        float g2 = float((rgbData[srcStride * (i + 1) + j] >> 10) & 0x3ff);
+        float b2 = float((rgbData[srcStride * (i + 1) + j] >> 20) & 0x3ff);
+
+        float r3 = float(rgbData[srcStride * (i + 1) + j + 1] & 0x3ff);
+        float g3 = float((rgbData[srcStride * (i + 1) + j + 1] >> 10) & 0x3ff);
+        float b3 = float((rgbData[srcStride * (i + 1) + j + 1] >> 20) & 0x3ff);
+
+        r0 /= 1023.0f;
+        g0 /= 1023.0f;
+        b0 /= 1023.0f;
+
+        r1 /= 1023.0f;
+        g1 /= 1023.0f;
+        b1 /= 1023.0f;
+
+        r2 /= 1023.0f;
+        g2 /= 1023.0f;
+        b2 /= 1023.0f;
+
+        r3 /= 1023.0f;
+        g3 /= 1023.0f;
+        b3 /= 1023.0f;
+
+        float y0 = coeffs[0] * r0 + coeffs[1] * g0 + coeffs[2] * b0;
+        float y1 = coeffs[0] * r1 + coeffs[1] * g1 + coeffs[2] * b1;
+        float y2 = coeffs[0] * r2 + coeffs[1] * g2 + coeffs[2] * b2;
+        float y3 = coeffs[0] * r3 + coeffs[1] * g3 + coeffs[2] * b3;
+
+        float u0 = (b0 - y0) / coeffs[3];
+        float u1 = (b1 - y1) / coeffs[3];
+        float u2 = (b2 - y2) / coeffs[3];
+        float u3 = (b3 - y3) / coeffs[3];
+
+        float v0 = (r0 - y0) / coeffs[4];
+        float v1 = (r1 - y1) / coeffs[4];
+        float v2 = (r2 - y2) / coeffs[4];
+        float v3 = (r3 - y3) / coeffs[4];
+
+        y0 = (y0 * 876.0f) + 64.0f + 0.5f;
+        y0 = CLIP3(y0, 64.0f, 940.0f);
+        yData[dst->stride[UHDR_PLANE_Y] * i + j] = uint16_t(y0) << 6;
+
+        y1 = (y1 * 876.0f) + 64.0f + 0.5f;
+        y1 = CLIP3(y1, 64.0f, 940.0f);
+        yData[dst->stride[UHDR_PLANE_Y] * i + j + 1] = uint16_t(y1) << 6;
+
+        y2 = (y2 * 876.0f) + 64.0f + 0.5f;
+        y2 = CLIP3(y2, 64.0f, 940.0f);
+        yData[dst->stride[UHDR_PLANE_Y] * (i + 1) + j] = uint16_t(y2) << 6;
+
+        y3 = (y3 * 876.0f) + 64.0f + 0.5f;
+        y3 = CLIP3(y3, 64.0f, 940.0f);
+        yData[dst->stride[UHDR_PLANE_Y] * (i + 1) + j + 1] = uint16_t(y3) << 6;
+
+        u0 = (u0 + u1 + u2 + u3) / 4;
+        v0 = (v0 + v1 + v2 + v3) / 4;
+
```

Comment:
Better to add a note "this is narrow-range"

Reply:
uhdr_raw_image_t contains a field for range. This value is configured to limited range. For yuv420p, this is configured to full range.